### PR TITLE
completes feature to add queries to get api/articles

### PR DESCRIPTION
**Pull request by caffreydev**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -578,3 +578,45 @@ describe('get /api/users', () => {
...
+  it('should respond appropriately to 3 queries and succesfully sort by specified collumn', () => {
+    return request(app)
+      .get('/api/articles?topic=mitch&sort_by=title&order=asc')
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.articles.length).toBe(12);
+        expect(body.articles).toBeSortedBy('title');
+      });
+  });
+
```
Comment (Jodiech) [inline __tests__/app.test.js:616]: I'd add another 200 test for a valid topic query that has no articles and responds with an empty array of articles 👀

File: app/controllers/articles.controllers.js [hunk 2]
```diff
@@ -25,8 +25,22 @@ exports.getArticleController = (req, res, next) => {
...
+  if (req.query.topic) {
+    topic = req.query.topic;
+  }
+  if (req.query.sort_by) {
+    sort_by = req.query.sort_by;
+  }
+  if (req.query.order) {
+    order = req.query.order;
+  }
+
```
Comment (Jodiech) [inline app/controllers/articles.controllers.js:42]: I'd consider having lines 29-41 in your models as this should be where all the data manipulation takes place. No need to change now but something to consider when refactoring in the future 👍

**Review by Jodiech — COMMENTED** [review]
Hey 👋 well done on this ticket! I'd like you to add another test to this and resubmit please :)

File: __tests__/app.test.js (hunk 1 above)
Comment (caffreydev) [reply __tests__/app.test.js:616]: fixed

File: __tests__/app.test.js [hunk 3]
```diff
@@ -578,3 +578,54 @@ describe('get /api/users', () => {
...
+  it('should respond appropriately with a 404 and message if queried with nonexistent collumn', () => {
+    return request(app)
+      .get('/api/articles?topic=chocolatecake')
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.message).toBe('topic chocolatecake not found');
+      });
+  });
+
```
Comment (pippaamy) [inline __tests__/app.test.js:625]: can we have an error test for invalid worthy

File: __tests__/app.test.js [hunk 4]
```diff
@@ -578,3 +578,54 @@ describe('get /api/users', () => {
...
+        expect(body.articles.length).toBe(12);
+        body.articles.forEach((article) => {
+          expect(article.topic).toBe('mitch');
+        });
+      });
+  });
+
+  it('should respond appropriately to 3 queries and succesfully sort by specified collumn', () => {
+    return request(app)
+      .get('/api/articles?topic=mitch&sort_by=title&order=asc')
```
Comment (pippaamy) [inline __tests__/app.test.js:597]: we need to check here that it is reacting to all the queries not just the sorted by

**Review by pippaamy — COMMENTED** [review]
Some more tests to add :)

File: __tests__/app.test.js (hunk 4 above)
Comment (caffreydev) [reply __tests__/app.test.js:597]: it already was - default is ascending, so it has switched to ascending.  also length being 12 means it has filtered out the one that isn't mitch. but have added in the test they all say mitch as a further validation
